gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,478 @@
+            res = checks.check_product_validity(product,
+                                                product_type=product_type)
+            assert res is None, (
+                f"Function returned {res} for product {product}, "
+                f"expected None"
+            )
+
+    # Test a product that doesn't exist
+    for product_type in product_types:
+        for product in [('IMERG'), 123, None]:
+            with pytest.raises(ValueError):
+                checks.check_product_validity(product,
+                                              product_type=product_type)
+            # Test a None product type
+            with pytest.raises(ValueError):
+                checks.check_product_validity(product, product_type=None)
+
+
+def test_check_time() -> None:
+    ''' Test that time is returned a datetime object from varying inputs '''
+
+    # Test a string
+    res = checks.check_time('2014-12-31')
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a datetime object
+    res = checks.check_time(datetime.datetime(2014, 12, 31))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_time(datetime.datetime(2014, 12, 31, 12, 30, 30, 300))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31, 12, 30, 30, 300)
+
+    # Test a np.datetime64 object of "datetime64[s]"
+    res = checks.check_time(np.datetime64('2014-12-31'))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a datetime.date
+    res = checks.check_time(datetime.date(2014, 12, 31))
+    assert isinstance(res, datetime.datetime)
+    assert res == datetime.datetime(2014, 12, 31)
+
+    # Test a non isoformat string
+    with pytest.raises(ValueError):
+        checks.check_time('2014/12/31')
+
+    # Test a non datetime object
+    with pytest.raises(TypeError):
+        checks.check_time(123)
+
+    # Check numpy multiple timestamp
+    with pytest.raises(ValueError):
+        checks.check_time(np.array(['2014-12-31', '2015-01-01'],
+                                   dtype='datetime64[s]'))
+
+    # Test with numpy non datetime64 object
+    with pytest.raises(ValueError):
+        checks.check_time(np.array(['2014-12-31']))
+
+
+def test_check_date() -> None:
+    ''' Check date/datetime objet is returned from varying inputs '''
+
+    # Test a datetime object
+    res = checks.check_date(datetime.datetime(2014, 12, 31))
+    assert isinstance(res, datetime.date)
+    assert res == datetime.date(2014, 12, 31)
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_date(datetime.datetime(2014, 12, 31, 12, 30, 30, 300))
+    assert isinstance(res, datetime.date)
+    assert res == datetime.date(2014, 12, 31)
+
+    # Test a none datetime object fails
```

Comment:
Can you improve the `check_date` function so that a numpy datetime or string are casted to `datetime.date`? 
We could actually call `check_time` and then to the output do `output.date` to get the `datetime.date object`